sympy/assumptions/tests/test_query.py

```diff
@@ -1857,6 +1979,7 @@ def test_real_pow():
     assert ask(Q.real(i**i), Q.imaginary(i)) is None  # i might be 2*I
     assert ask(Q.real(x**i), Q.imaginary(i)) is None  # x could be 0
     assert ask(Q.real(x**(I*pi/log(x))), Q.real(x)) is True
+    assert ask(Q.real(x**y), Q.nonnegative(x) & x == y) is None
```

Comment:
x == y always evaluates to False. == does structural equality in SymPy. To get an unevaluated boolean, use `Eq`.

Reply:
Also `&` has a higher precedence than `==` e.g.:
```
In [7]: x & y == y                                                                                                                
Out[7]: False

In [8]: (x & y) == y                                                                                                              
Out[8]: False

In [9]: x & (y == y)                                                                                                              
Out[9]: x
```